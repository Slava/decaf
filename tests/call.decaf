void f () {
  f();
  f(1);
  f(1, 2, 3);
  f(f(1), f(f(2, 3)));
  this.something.f();
  this.is.another[one()](1, 2);
}

void builtins () {
  Print(1);
  Print(1,2,1+2+3);
  ReadLine();
  new SomeClass;
  NewArray(123, void);
  NewArray(123+5, int);
}

