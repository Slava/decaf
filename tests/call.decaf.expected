Program
  Function void: f:
    (#f)
    (#f 1)
    (#f 1 2 3)
    (#f (#f 1) (#f (#f 2 3)))
    (<<this>.<#something>>.<#f>)
    (<<<this>.<#is>>.<#another>>[(#one)] 1 2)
